File: BankManagementSystem-backend/src/main/java/com/bank/model/Account.java
package com.bank.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;

@Data
@Entity
public class Account 
{
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long accountId;
	private String bankName;
	private String branchArea;
	private String branchCity;
	private String branchState;
	private String branchpincode;
	private String managername;
	private double Balance;
	private String type;
	
	
	@ManyToOne
    @JoinColumn(name = "customer_id")
	@JsonIgnore
    private Customer customer;
	
	// Method to deposit funds
    public void deposit(double amount) {
        this.Balance += amount;
    }

    // Method to withdraw funds
    public boolean withdraw(double amount) {
        if (this.Balance >= amount) {
            this.Balance -= amount;
            return true; // Withdrawal successful
        }
        return false; // Insufficient funds
    }

    
  
    
   
    
}

File: BankManagementSystem-frontend/BankManagementSystem-Angular/src/app/account.ts
export class Account {
    accountId:number;
    bankName:string;
    branchArea:string;
    branchCity:string;
    branchState:string;
    branchpincode:string;
    managername:string;
    balance:number;
    type:string;
    fromAccountId:number;
    toAccountId:number;

    constructor(
        accountId:number,
        bankName:string,
        branchArea:string,
        branchCity:string,
        branchState:string,
        branchpincode:string,
        managername:string,
        balance:number,
        type:string,
        fromAccountId:number,
        toAccountId:number
    ){
        this.accountId=accountId;
        this.bankName=bankName;
        this.branchArea=branchArea;
        this.branchCity=branchCity;
        this.branchState=branchState;
        this.branchpincode=branchpincode;
        this.managername=managername;
        this.balance=balance;
        this.type=type;
        this.fromAccountId=fromAccountId;
        this.toAccountId=toAccountId;
    }

  }
  

